### ISSUE TITLE
Implement self-heal to fix block volumes who's operation failed halfway

### ISSUE THREAD

**Issue by pkalever** [gluster/gluster-block#167]
More details later

**Comment by pkalever** [comment]
@lxbsz please use this issue, for adding details about the detailed design. Thanks!

**Comment by lxbsz** [comment]
> @lxbsz please use this issue, for adding details about the detailed design. Thanks!

@pkalever 
Sure, I have gone through the design doc and also the selfheal design in glusterfs AFR. Should begin try to code this next week.

Thanks.

**Comment by pkalever** [comment]
> > @lxbsz please use this issue, for adding details about the detailed design. Thanks!
> 
> @pkalever
> Sure, I have gone through the design doc and also the selfheal design in glusterfs AFR. Should begin try to code this next week.

@lxbsz 

The current design doc is missing basic details like for example, if we are maintaining the details about the block volumes list which needs healing in the block hosting volume, then where actually we maintain the complete list of such block hosting volumes, so that the self-heal thread get the details of block volumes needing heal ?

Also it has many unaddressed comments left. Hence I request to create a separate design document first, before starting to code.

Lets get approval from others on the proposed design and then fallback to implementation.

Thanks!

**Comment by lxbsz** [comment]
> > > @lxbsz please use this issue, for adding details about the detailed design. Thanks!
> > 
> > 
> > @pkalever
> > Sure, I have gone through the design doc and also the selfheal design in glusterfs AFR. Should begin try to code this next week.
> 
> @lxbsz
> 
> The current design doc is missing basic details like for example, if we are maintaining the details about the block volumes list which needs healing in the block hosting volume, then where actually we maintain the complete list of such block hosting volumes, so that the self-heal thread get the details of block volumes needing heal ?

Yeah, IMO, sometimes maybe we can just try to code and it will be more clear for the basic detail then.

> 
> Also it has many unaddressed comments left. Hence I request to create a separate design document first, before starting to code.
> 
> Lets get approval from others on the proposed design and then fallback to implementation.
> 
No problem.
Thanks.

**Comment by pkalever** [comment]
Won't Fix, closing now.
